internal/subscriber/retry/retrier_test.go

```diff
@@ -87,3 +104,71 @@ func Test_Interval_Calculator(t *testing.T) {
 		assert.Equal(t, test.want, actualIntervals)
 	}
 }
+
+func TestRetrier_Start(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	ctxWt, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	mockBrokerStore := mocks.NewMockIBrokerStore(ctrl)
+	mockCache := mocks2.NewMockICache(ctrl)
+	mockCache.EXPECT().Get(gomock.AssignableToTypeOf(ctxWt), gomock.Any()).Return([]byte{'0'}, nil).AnyTimes()
+	dcs := getMockDelayConsumer(ctx, ctrl)
+	producer := getMockProducer(ctx, ctrl)
+	mockBrokerStore.EXPECT().GetConsumer(
+		ctxWt,
+		gomock.Any(),
+	).Return(dcs, nil).AnyTimes()
+	mockBrokerStore.EXPECT().RemoveConsumer(ctxWt, gomock.Any()).AnyTimes()
+	mockBrokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+
+	r := &Retrier{
+		subscriberID:   "subscription-id",
+		subs:           getMockSubscriptionModel(),
+		bs:             mockBrokerStore,
+		ch:             mockCache,
+		backoff:        NewExponentialWindowBackoff(),
+		finder:         NewClosestIntervalWithCeil(),
+		handler:        NewPushToPrimaryRetryTopicHandler(mockBrokerStore),
+		delayConsumers: sync.Map{},
+		errChan:        make(chan error),
+	}
+	expDelayTopics := r.subs.GetDelayTopics()
+	sort.Strings(expDelayTopics)
```

Comment:
these are already sorted.. why are we sorting it again..?

Reply:
@sakshi-goyal-razorpay these are sorted according to delay time but when sort.Strings(dcTopics) is run it will sort according to strings.
[projects/test-project/topics/test-subscription.delay.150.seconds projects/test-project/topics/test-subscription.delay.1800.seconds projects/test-project/topics/test-subscription.delay.30.seconds projects/test-project/topics/test-subscription.delay.300.seconds projects/test-project/topics/test-subscription.delay.3600.seconds projects/test-project/topics/test-subscription.delay.5.seconds projects/test-project/topics/test-subscription.delay.60.seconds projects/test-project/topics/test-subscription.delay.600.seconds]
[projects/test-project/topics/test-subscription.delay.5.seconds projects/test-project/topics/test-subscription.delay.30.seconds projects/test-project/topics/test-subscription.delay.60.seconds projects/test-project/topics/test-subscription.delay.150.seconds projects/test-project/topics/test-subscription.delay.300.seconds projects/test-project/topics/test-subscription.delay.600.seconds projects/test-project/topics/test-subscription.delay.1800.seconds projects/test-project/topics/test-subscription.delay.3600.seconds]

Reply:
If they're already sorted by delay time then assert should work right?

Reply:
@vnktram r.delayConsumers will give array of delaytopics which is unordered so sort it but this will sorted according to alphabet which will be in order : 
[projects/test-project/topics/test-subscription.delay.150.seconds projects/test-project/topics/test-subscription.delay.1800.seconds projects/test-project/topics/test-subscription.delay.30.seconds projects/test-project/topics/test-subscription.delay.300.seconds projects/test-project/topics/test-subscription.delay.3600.seconds projects/test-project/topics/test-subscription.delay.5.seconds projects/test-project/topics/test-subscription.delay.60.seconds projects/test-project/topics/test-subscription.delay.600.seconds]

but r.subs.getDelayTopics() will give this : 
[projects/test-project/topics/test-subscription.delay.5.seconds projects/test-project/topics/test-subscription.delay.30.seconds projects/test-project/topics/test-subscription.delay.60.seconds projects/test-project/topics/test-subscription.delay.150.seconds projects/test-project/topics/test-subscription.delay.300.seconds projects/test-project/topics/test-subscription.delay.600.seconds projects/test-project/topics/test-subscription.delay.1800.seconds projects/test-project/topics/test-subscription.delay.3600.seconds]

so i need to sort that while compairing